### ISSUE TITLE
Make auto close html smarter

### ISSUE THREAD

**Issue by adambuczynski** [mattberkowitz/autoclose-html#155]
Given the amount of open issues I don't have high hopes for this being addressed, but...

```html
<p class="text-danger" ng-message="required">Checkbox is required.</p>
```

If I accidentally erase the closing `>` before `Checkbox`, e.g.:

```html
<p class="text-danger" ng-message="required"Checkbox is required.</p>
```

and then add it in again, autoclose-html turns it into this wonderful HTML:

```html
<p class="text-danger" ng-message="required">
                
</p>Checkbox is required.</p>
```

Can we make it a bit smarter to check if a matching closing tag is already in place, and then just leave the HTML alone?

**Comment by mattberkowitz** [comment]
This is a dupe of #108 so I'm going to close it. Would love to have this functionality, but it's a hairier problem than it seems, as it requires knowledge of which tags are currently open as well as scanning forward in the document and arbitrary distance to see what closing tags there are.

You quickly get into the weeds with things like:
* If you're dealing with a `<div>` within a `<div>` which was the closing tag intended for?
* What if there are malformed tags, the current regex based tag recognition system isn't really great for things like this (or really ideal for html parsing in general)

Long story short, this would be a great feature to have, but it's unlikely something I will personally be able to address in the short term

**Comment by adambuczynski** [comment]
@mattberkowitz thanks for the clarification.

I appreciate the difficulties involved, but I think some relatively simple rules could go a long way. For example, simply find a matching tag on the same line. This is most likely to be the correct matching tag. Not always, true, but 95% of the time it would be.

I'll subscribe to issue 108.
